bot/exts/events/advent_of_code/views/dayandstarview.py

```diff
@@ -0,0 +1,68 @@
+
+import discord
+
+AOC_DAY_ANDSTAR_TEMPLATE = "{rank: >4} | {name:25.25} | {completion_time: >10}"
+
+
+class AoCDropdownView(discord.ui.View):
+    """Represents a Dropdown to filter AoC stats by Day and Star."""
+
+    def __init__(self, original_author: discord.Member, day_and_star_data: dict[str: dict], maximum_scorers: int):
+        super().__init__()
+        self.day = 1
+        self.star = 1
+        self.data = day_and_star_data
+        self.element_map = {element.custom_id: element for element in self.children}
+        self.maximum_scorers = maximum_scorers
+        self.original_author = original_author
+
+    def generate_output(self) -> str:
+        """Generates a formatted codeblock with AoC statistics based on the currently selected day and star."""
+        header = AOC_DAY_ANDSTAR_TEMPLATE.format(rank="Rank", name="Name",
+                                                 completion_time="Completion time (UTC)")
+        header = f"{header}\n{'-' * (len(header) + 2)}"
+        lines = [header]
+
+        for rank, scorer in enumerate(self.data[f"{self.day}-{self.star}"][:self.maximum_scorers]):
+            time_data = datetime.fromtimestamp(scorer['completion_time']).strftime("%I:%M:%S %p")
+            lines.append(AOC_DAY_ANDSTAR_TEMPLATE.format(datastamp="", rank=rank + 1, name=scorer['member_name'],
+                                                         completion_time=time_data))
+        joined_lines = "\n".join(lines)
+        return f"Statistics for Day: {self.day}, Star: {self.star}.\n ```\n{joined_lines}\n```"
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        """A global check to ensure that the interacting user is the user who invoked the command originally."""
+        return interaction.user == self.original_author
+
+    @discord.ui.select(
+        placeholder="Day",
+        options=[discord.SelectOption(label=str(i)) for i in range(1, 26)],
+        custom_id="day_select"
+    )
+    async def day_select(self, select: discord.ui.Select, interaction: discord.Interaction) -> None:
+        """Represents a Dropdown to choose a Day of the AoC."""
+        self.day = select.values[0]
+        return
+
+    @discord.ui.select(
+        placeholder="Star",
+        options=[discord.SelectOption(label=str(i)) for i in range(1, 3)],
+        custom_id="star_select"
+    )
+    async def star_select(self, select: discord.ui.Select, interaction: discord.Interaction) -> None:
+        """Represents a Dropdown to choose either the first or the second star."""
+        self.star = select.values[0]
+        return
+
+    @discord.ui.button(label="Fetch", style=discord.ButtonStyle.blurple)
+    async def fetch(self, button: discord.ui.Button, interaction: discord.Interaction) -> None:
+        """Represents a button that fetches the statistics based on the dropdown values."""
+        if not self.element_map['star_select'].values or not self.element_map['day_select'].values:
```

Comment:
Can't we just use `self.star` and `self.day`?

Reply:
I don't think so, no. This check is meant to check whether in the current interaction in the very moment when this button is pushed there is one value chosen from both of the dropdowns, since when the user pushes the fetch button, the interaction will revert itself to its original form. (This way we ensure that the placeholder will be shown every time they want to interact with the view). If I were to use those variables in this check,  the view would just fetch the same view that was fetched previously without any warning.

Reply:
This doesn't even work, I can select one of the dropdowns and fetch. I think it is unneccessary.